arches/settings_utils.py

```diff
@@ -0,0 +1,140 @@
+    arches_applications -- tuple of installed arches_app names
+    additional_directories -- list of os-safe absolute paths
+    context_processors -- list of strings representing desired context processors
+    """
+    directories = []
+    
+    if additional_directories:
+        for additional_directory in additional_directories:
+            directories.append(additional_directory)
+
+    if app_root:
+        directories.append(os.path.join(app_root, "templates"))
+
+    if arches_applications:
+        site_package_path = site.getsitepackages()[0]
+
+        for arches_app in arches_applications:
+            arches_app_path = os.path.join(site_package_path, arches_app)
+
+            if os.path.exists(arches_app_path):
+                directories.append(os.path.join(arches_app_path, "templates"))
+            else:
+                egg_link_path = arches_app_path.replace('_', '-')
+                egg_link_path += '.egg-link'
+
+                if os.path.exists(egg_link_path):
+                    original_path = Path(egg_link_path).read_text()
+                    original_path = original_path.replace('.', '')
+                    original_path = original_path.strip()
+                    directories.append(os.path.join(original_path, arches_app, 'templates'))
+
+    directories.append(os.path.join(root_dir, "app", "templates"))
+
+    return [
+        {
+            "BACKEND": "django.template.backends.django.DjangoTemplates",
+            "DIRS": directories,
+            "APP_DIRS": True,
+            "OPTIONS": {
+                "context_processors": context_processors
+                if context_processors
+                else [
+                    "django.contrib.auth.context_processors.auth",
+                    "django.template.context_processors.debug",
+                    "django.template.context_processors.i18n",
+                    "django.template.context_processors.media",
+                    "django.template.context_processors.static",
+                    "django.template.context_processors.tz",
+                    "django.template.context_processors.request",
+                    "django.contrib.messages.context_processors.messages",
+                    "arches.app.utils.context_processors.livereload",
+                    "arches.app.utils.context_processors.map_info",
+                    "arches.app.utils.context_processors.app_settings",
+                ],
+                "debug": debug,
+            },
+        },
+    ]
+
+
+def transmit_webpack_django_config(
+    root_dir, app_root, static_url, public_server_address, webpack_development_server_port, arches_applications=None
+):
+    print(
+        json.dumps(
+            {
+                "APP_ROOT": app_root,
+                "ARCHES_APPLICATIONS": list(arches_applications) if arches_applications else [],
+                "ARCHES_APPLICATIONS_PATH": site.getsitepackages()[0],
```

Comment:
When I `arches-project create`'d a new arches project, and then pip installed the modularized version of afs (i.e. much like your [new repo](https://github.com/archesproject/arches_for_science_package)), all is groovy.

When I pip install with `-e` for editable mode, though, even with `--config-settings editable_mode=compat`, I get:

```py
  File "/Users/jwalls/prj/venv39/lib/python3.9/site-packages/arches/management/commands/packages.py", line 932, in load_package
    raise Exception("this is an invalid package source")
Exception: this is an invalid package source
```

Upon debugging, it's looking for a project path at 
```
/Users/jwalls/prj/venv39/lib/python3.9/site-packages/{ARCHES_APPLICATION}/pkg
```
which doesn't exist, because I used `-e`. Instead I have a `.dist-info` folder and a `direct_url.json` in there that can be parsed to get the path to the files.

I noticed that setuptools has a bunch of [disclaimers](https://setuptools.pypa.io/en/latest/userguide/development_mode.html#limitations) about editable mode, e.g. "There is no guarantee that the editable installation will be performed using a specific technique." so unfortunately we may need to tread carefully here!

Maybe this needs to be a follow-up card?

Reply:
should be fixed in latest